contracts/stake-cw20-external-rewards/src/contract.rs

```diff
@@ -239,23 +245,28 @@ pub fn update_rewards(deps: &mut DepsMut, env: &Env, addr: &Addr) -> StdResult<(
     )?;
     PENDING_REWARDS.update::<_, StdError>(deps.storage, addr.clone(), |r| {
         Ok(r.unwrap_or_default() + earned_rewards)
-    });
+    })?;
 
-    USER_REWARD_PER_TOKEN.save(deps.storage, addr.clone(), &reward_per_token);
+    USER_REWARD_PER_TOKEN.save(deps.storage, addr.clone(), &reward_per_token)?;
     LAST_UPDATE_BLOCK.save(deps.storage, &env.block.height)?;
-    Ok({})
+    Ok(())
 }
 
 pub fn get_reward_per_token(deps: Deps, env: &Env, staking_contract: &Addr) -> StdResult<Uint256> {
     let reward_config = REWARD_CONFIG.load(deps.storage)?;
     let total_staked = get_total_staked(deps, staking_contract)?;
-    let current_block = min(env.block.height, reward_config.periodFinish);
+    let current_block = min(env.block.height, reward_config.period_finish);
     let last_update_block = LAST_UPDATE_BLOCK.load(deps.storage).unwrap_or_default();
     let prev_reward_per_token = REWARD_PER_TOKEN.load(deps.storage).unwrap_or_default();
     let additional_reward_per_token = if total_staked == Uint128::zero() {
         Uint256::zero()
     } else {
-        let numerator = Uint256::from(reward_config.rewardRate.checked_mul(Uint128::from(current_block - last_update_block))?).checked_mul(scale_factor())?;
+        let numerator = Uint256::from(
+            reward_config
+                .reward_rate
+                .checked_mul(Uint128::from(current_block - last_update_block))?,
```

Comment:
This will still overflow even though it eventually becomes a `Uint256`. Probably want to use `full_mul` here.

Reply:
Great suggestion. Fixed